### ISSUE TITLE
System.ArgumentOutOfRangeException from Script Execution

### ISSUE THREAD

**Issue by AnyOldName3** [erri120/OMODFramework#35]
**Describe the bug**
The animation installer OMOD script behaves inconsistently between OBMM and OMODFramework. This kills the OMOD.

**To Reproduce**
* Download the 7z for non-Wrye Bash users.
* Extract it.
* Attempt to run the script in the animations installer OMOD with OMODFramework.
* Observe a `System.ArgumentOutOfRangeException` when the script reaches line 87.

**Versions**
OMODFramework-Version: 2.0.1, master
.NET Framework-Version: `.NET 4.8`

**Additional context**
The script uses string functions to emulate arrays. I don't believe all the string functions behave the same way, although it could be that string interpolation isn't behaving consistently, too.

The offending line is
```OBMMScript
RemoveString vMapName "%vMapName%" 0 %vLen%
```

Using a tweaked version of the OMOD, adding the line 
```OBMMScript
Message "Line is 'RemoveString vMapName \"%vMapName%\" 0 %vLen%'"
```
produces different results in OBMM and OMODFramework.

With OBMM: 
![image](https://user-images.githubusercontent.com/8485112/90337350-d30d8f80-dfd9-11ea-90d1-2c52daa09e7c.png)
With OMODFramework: 
![image](https://user-images.githubusercontent.com/8485112/90337355-dbfe6100-dfd9-11ea-901e-91a314e268e0.png)

**Comment by erri120** [comment]
`RemoveString vMapName "%vMapName%" 0 %vLen%` with `vMapName` being `GetFileName vMapName "%vMapName%"` in a `For Each DataFile vMapName "%vKFmaps%"`.
Enumeration might differ from OBMM and OMODFramework thus the different message results.

**Comment by AnyOldName3** [comment]
More information: The script builds a big select statement as a string and executes it with `ExecLines`. In OBMM, that string has a lot fewer backslashes in it than in OMODFramework:

OBMM:
![image](https://user-images.githubusercontent.com/8485112/90338649-fbe65280-dfe2-11ea-9953-8a7824900dff.png)
OMODFramework:
![image](https://user-images.githubusercontent.com/8485112/90338652-030d6080-dfe3-11ea-8513-0ad2d8a94990.png)

This Select block is what sets the paths that get iterated over, so it's possible that due to the string differences, none of the cases match the selection, and the variable never gets set.

**Comment by AnyOldName3** [comment]
Yep, none of the cases are matching. I changed it to
```OBMMScript

	SetVar vCases "%vCases%\	
		Case %vCfolder%%NewLine%\
			SetVar vKFpath \"%vKFspath%\\\\\\\\%vCfolder%\"%NewLine%\
			SetVar vKFmaps \"%vKFmpath%\\\\\\\\%vCfolder%\"%NewLine%\
			Message \"matched %vCfolder%\"%NewLine%\
		Break%NewLine%"
```
and it prints that it matched something in OBMM, but not in OMODFramework.

**Comment by AnyOldName3** [comment]
I think this might be as simple as `vCfolder` starting with a backslash in OMODFramework, but not OBMM, and these being eaten by something later on for `Select` but not for `Case`.

**Comment by AnyOldName3** [comment]
Removing leading directory separator characters here gets past the issue, but I'm seeing other problems later on.

https://github.com/erri120/OMODFramework/blob/master/OMODFramework.Scripting/Scripting/OBMMScriptHandler.cs#L833

**Comment by AnyOldName3** [comment]
The issue later on seems to be that `for each DataFile` still searches for directories. There also seem to be more backslashes than are strictly necessary, but I don't have any evidence to say they're not present in OBMM as they're probably benign.

**Comment by AnyOldName3** [comment]
I've got it working locally. This script is insane! I'm pretty sure if Timeslip had expected anyone to write this script, there'd be a bunch more functions in the API to make it simpler. It's pretty conclusive proof of how little a language really needs to express any computation.

**Comment by AnyOldName3** [comment]
There are issues with the other OMOD in the mod, too. It seems to be running the `CopyDataFile` implementation when `CopyPlugin` gets called.

**Comment by AnyOldName3** [comment]
`var plugin = line.ElementAt(0).StartsWith("Plugin");` obviously won't be true for `"CopyPlugin"`.

**Comment by AnyOldName3** [comment]
Resolved by #36
